diff --git a/lib/main.js b/lib/main.js
index 6ed0c2d..7f0b6b0 100644
--- a/lib/main.js
+++ b/lib/main.js
@@ -1020,34 +1020,7 @@
           sendRequest(refs, request2, (error2) => {
             if (error2) reject(new Error(error2));
             else resolve(void 0);
           });
         }),
-        serve: (options2 = {}) => new Promise((resolve, reject) => {
-          if (!streamIn.hasFS) throw new Error(`Cannot use the "serve" API in this environment`);
-          const keys = {};
-          const port = getFlag(options2, keys, "port", mustBeInteger);
-          const host = getFlag(options2, keys, "host", mustBeString);
-          const servedir = getFlag(options2, keys, "servedir", mustBeString);
-          const keyfile = getFlag(options2, keys, "keyfile", mustBeString);
-          const certfile = getFlag(options2, keys, "certfile", mustBeString);
-          const fallback = getFlag(options2, keys, "fallback", mustBeString);
-          const onRequest = getFlag(options2, keys, "onRequest", mustBeFunction);
-          checkForInvalidFlags(options2, keys, `in serve() call`);
-          const request2 = {
-            command: "serve",
-            key: buildKey,
-            onRequest: !!onRequest
-          };
-          if (port !== void 0) request2.port = port;
-          if (host !== void 0) request2.host = host;
-          if (servedir !== void 0) request2.servedir = servedir;
-          if (keyfile !== void 0) request2.keyfile = keyfile;
-          if (certfile !== void 0) request2.certfile = certfile;
-          if (fallback !== void 0) request2.fallback = fallback;
-          sendRequest(refs, request2, (error2, response2) => {
-            if (error2) return reject(new Error(error2));
-            if (onRequest) {
-              requestCallbacks["serve-request"] = (id, request3) => {
-                onRequest(request3.args);
-                sendResponse(id, {});
-              };
-            }
-            resolve(response2);
-          });
-        }),
+        serve: () => Promise.reject(new Error("The esbuild serve API is disabled in this project due to GHSA-67mh-4wv8-2f99.")),
         cancel: () => new Promise((resolve) => {
           if (didDispose) return resolve();
           const request2 = {
             command: "cancel",
             key: buildKey
